tests/experimental/test_bundles.py

```diff
@@ -0,0 +1,261 @@
+        )
+
+        bundle = create_bundle_for_flow_run(flow_with_parameters, flow_run)
+        process = execute_bundle_in_subprocess(bundle)
+
+        process.join()
+        assert process.exitcode == 0
+
+        flow_run = await self.get_flow_run_for_flow(
+            prefect_client, flow_with_parameters.name
+        )
+        assert flow_run.state is not None
+        assert flow_run.state.is_completed()
+        assert await flow_run.state.result() == "x: 42, y: hello"
+
+    async def test_with_provided_context(
+        self, prefect_client: PrefectClient, engine_type: Literal["sync", "async"]
+    ):
+        if engine_type == "sync":
+
+            @flow(
+                name="TestExecuteBundleInSubprocess.test_with_provided_context[sync]",
+                persist_result=True,
+            )
+            def context_flow():
+                return TagsContext.get().current_tags
+        else:
+
+            @flow(
+                name="TestExecuteBundleInSubprocess.test_with_provided_context[async]",
+                persist_result=True,
+            )
+            async def context_flow():
+                return TagsContext.get().current_tags
+
+        context = {"tags_context": TagsContext(current_tags={"foo", "bar"}).serialize()}
+
+        flow_run = await prefect_client.create_flow_run(
+            flow=context_flow,
+        )
+
+        bundle = create_bundle_for_flow_run(
+            flow=context_flow, flow_run=flow_run, context=context
+        )
+        process = execute_bundle_in_subprocess(bundle)
+        process.join()
+        assert process.exitcode == 0
+
+        flow_run = await self.get_flow_run_for_flow(prefect_client, context_flow.name)
+        assert flow_run.state is not None
+        assert flow_run.state.is_completed()
+        assert await flow_run.state.result() == {"foo", "bar"}
+
+    async def test_flow_is_aborted(
+        self, engine_type: Literal["sync", "async"], prefect_client: PrefectClient
+    ):
+        if engine_type == "sync":
+
+            @flow(
+                name="TestExecuteBundleInSubprocess.test_flow_is_aborted[sync]",
+                persist_result=True,
+            )
+            def foo():
+                raise Abort()
+        else:
+
+            @flow(
+                name="TestExecuteBundleInSubprocess.test_flow_is_aborted[async]",
+                persist_result=True,
+            )
+            async def foo():
+                raise Abort()
+
+        flow_run = await prefect_client.create_flow_run(flow=foo)
+
+        bundle = create_bundle_for_flow_run(foo, flow_run)
+        process = execute_bundle_in_subprocess(bundle)
+        process.join()
+        assert process.exitcode == 0
+
+        flow_run = await prefect_client.read_flow_run(flow_run.id)
+        # Stays in running state because the flow run is aborted manually
+        assert flow_run.state is not None
+        assert flow_run.state.is_running()
```

Comment:
hmm, I'm likely missing something - why this is expected?

Reply:
The flow raises an abort, which causes the flow to stop running immediately without any state changes. This test ensures that the process has a 0 exit code in this case and I wanted to check to make sure the flow entered running before aborting.